### PR TITLE
feat(templates): add interaction handler templates

### PR DESCRIPTION
**Pull request by devramsean0**
Title says it all,
all TS ones are tested and the JS ones are from the guide

### CODE REVIEW

**Review by enxg — CHANGES_REQUESTED** [review]
Please add newlines to the end of the files.

File: templates/components/autocompleteinteractionhandler.js.sapphire [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+	"category": "interaction-handlers",
+}
+---
+const { InteractionHandler, InteractionHandlerTypes } = require('@sapphire/framework');
```
Comment (favna) [inline templates/components/autocompleteinteractionhandler.js.sapphire:5]: For all template files use the Sapphire prettier config to format them. Copy the template code to [the prettier playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6VACAwhAJnDACzgCc4QAaECABxgEtoBnZUAQxJIgHcAFDhCxRsANtzYBPFlQBGJNmADWcGAGU2AWzgAZelHJIAZqKZxZ8pStU0FegObIYJAK5mQcDTLi58ubWyg7ZzY7OAAxCBINNhgGQOQQNmcYCEoQQhgNEQB1Qnp4JhswOFVBfPoAN3yJBLAmaRA9UxIYXnk7aORjEVMqACsmAA9VexE4AEVnCHgukzcbEmaEmAkaOCYwEno6NJot2Gz6XBhCZABGAFYABio9iFNs+RoEvfXSCvIqAEcp+DbaISJJgAWn03m8aTIP3oZDaIU6RjmVFMGnojhcbiYowmvwMTlcVBgbBkh2OpyQABZCfJ6CJ7DgNAiQFBoJ8QM5TAAVYlCfFuCquACSUHwsFUm22MAAgiLVCsxrMenAAL7KoA) then paste the result back.

Also for all JS template files please add JSDoc comments similar to the already existing ones to add some basic types to JS code

**Review by favna — CHANGES_REQUESTED** [review]

File: templates/components/autocompleteinteractionhandler.js.sapphire (hunk 1 above)
Comment (devramsean0) [reply templates/components/autocompleteinteractionhandler.js.sapphire:5]: So adding
```
	/**
		* @param {InteractionHandler.Context} context
	*/
``` to every JS interaction-handler template

File: templates/components/autocompleteinteractionhandler.ts.sapphire [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+		// Do Whatever
+		await interaction.editReply({ content: 'Hello World' });
+	}
+
+	public override async parse(interaction: AutocompleteInteraction) {
+		// Make sure the interaction you want is what you get
+		if (!interaction.customId.startsWith('{{name}}')) return this.none();
+		await interaction.deferReply();
+		return this.some();
+	}
```
Comment (vladfrangu) [inline templates/components/autocompleteinteractionhandler.ts.sapphire:23]: This should do the same as the JS template!

File: templates/components/autocompleteinteractionhandler.ts.sapphire [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}
+---
+import { ApplyOptions } from '@sapphire/decorators';
+import { InteractionHandler, InteractionHandlerTypes } from '@sapphire/framework';
+import type { AutocompleteInteraction } from 'discord.js';
+
+@ApplyOptions<InteractionHandler.Options>({
+	type: InteractionHandlerTypes.Autocomplete
+})
+export default class extends InteractionHandler {
```
Comment (vladfrangu) [inline templates/components/autocompleteinteractionhandler.ts.sapphire:12]: We don't do export defaults in other templates

File: templates/components/buttoninteractionhandler.js.sapphire [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	}
+
+	async run(interaction) {
+		await interaction.reply({
+			content: 'Hello from a button interaction handler!',
+			// Let's make it so only the person who pressed the button can see this message!
+			ephemeral: true
+		});
+	}
+}
```
Comment (vladfrangu) [inline templates/components/buttoninteractionhandler.js.sapphire:29]: Formatting is not done

File: templates/components/buttoninteractionhandler.js.sapphire [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+		* @param {InteractionHandler.Context} context
+	*/
+	constructor(ctx, options) {
+		super(ctx, {
+			...options,
+			interactionHandlerType: InteractionHandlerTypes.Button
+		});
+	}
+
+	parse(interaction) {
```
Comment (vladfrangu) [inline templates/components/buttoninteractionhandler.js.sapphire:17]: Let's try to be consistent with the parse vs run location in the files!

File: templates/components/modalinteractionhandler.js.sapphire [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+		return this.some();
+	}
+
+	async run(interaction) {
+		await interaction.reply({
+			content: 'Thank you for submitting the form!',
+			ephemeral: true
+		});
+	}
+}
```
Comment (vladfrangu) [inline templates/components/modalinteractionhandler.js.sapphire:28]: Ditto

File: templates/components/selectmenuinteractionhandler.js.sapphire [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+		return this.some();
+	}
+
+	async run(interaction) {
+		await interaction.reply({
+			// Remember how we can have multiple values? Let's get the first one!
+			content: `You selected: ${interaction.values[0]}`
+		});
+	}
+}
```
Comment (vladfrangu) [inline templates/components/selectmenuinteractionhandler.js.sapphire:28]: Ditto

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: templates/components/autocompleteinteractionhandler.js.sapphire (hunk 1 above)
Comment (favna) [reply templates/components/autocompleteinteractionhandler.js.sapphire:5]: Yes

**Comment by Swiizyy** [comment]
To be closed has been moved to #216

**Pull request closed without merging**